src/main/java/com/google/devtools/build/lib/bazel/commands/VendorCommand.java

```diff
@@ -398,6 +400,14 @@ private void vendor(CommandEnvironment env, ImmutableList<RepositoryName> reposT
             .getOutputBase()
             .getRelative(LabelConstants.EXTERNAL_REPOSITORY_LOCATION);
     vendorManager.vendorRepos(externalPath, reposToVendor);
+
+    // 3. Invalidate RepositoryDirectoryValue for vendored repos.
+    for (RepositoryName repo : reposToVendor) {
```

Comment:
We can do this with a single traversal by checking containment in `reposToVendor` inside the predicate.

Reply:
Ah, right! 😅